File: python_algorithm/programmers/x만큼간격이있는n개의숫자.py
# 문제설명
# 함수 solution은 정수 x와 자연수 n을 입력 받아, 
# x부터 시작해 x씩 증가하는 숫자를 n개 지니는 리스트를 리턴해야 합니다.
# 다음 제한 조건을 보고, 조건을 만족하는 함수, solution을 완성해주세요.

# 제한조건
# 1. x는 -10000000 이상, 10000000 이하인 정수입니다.
# 2. n은 1000 이하 자연수입니다.

# 입출력예
# x = 2, n =5
# answer = [2, 4, 6, 8, 10]
# x = 4, n =3
# answer = [4, 8, 12]
# x = -4, n =2
# answer = [-4, -8]

def solution(x, n):
    if x > 0:
        answer = [i for i in range(x, n*x+1, x)]
    elif x == 0:
        answer = [0]*n
    else:
        answer = [i for i in range(x, n*x-1, x)]
    
    return answer


def init():
    x, n = map(int, input().split())
    print(solution(x, n))

init()

File: js_algorithm/programmers/x만큼간격이있는n개의숫자.js
// 내가 푼 코드 반복문 활용
function solution(x, n) {
  const answer = [];
  for (let i = 1; i <= n; i++) {
    answer.push(x * i);
  }
  return answer;
}

// 다른 사람 코드 map 함수 활용.
// Array객체를 활요해서 빈 배열 생성 후
// fill매서드로 x값 할당
// map 함수의 value와 index를 활용해 각각의 값을 변경
function solution2(x, n) {
  const answer = Array(n)
    .fill(x)
    .map((value, index) => (index + 1) * value);
  return answer;
}

function init() {
  x = [2, 4, -4];
  n = [5, 3, 2];
  for (let i = 0; i < x.length; i++) {
    // console.log(solution(x[i], n[i]));
    console.log(solution2(x[i], n[i]));
  }
}

init();
